### ISSUE TITLE
Feature: Ignore certain URLs or URL patterns when auto-save is turned on

### ISSUE THREAD

**Issue by mpeteuil** [gildas-lormeau/SingleFile#528]
**Is your feature request related to a problem? Please describe.**
I want to auto-save everything in general, but there are certain websites that I don't want to be auto-saved. I haven't been able to find a way in SingleFile to turn on auto-save but then have exceptions like that.

**Describe the solution you'd like**
I'd like to be able to provide a list of URLs or perhaps regular expression where if the current URL matches it will be ignored by auto-save instead of saved.

**Describe alternatives you've considered (optional)**
A less-than-ideal work-around I've considered is watching the filesystem location where SingleFile is saving the pages and deleting the ones I would have wanted to ignore. There are a few problems with this though including how to identify them once they're saved and for anything that might be highly sensitive you might not want it saved to disk at all in the first place since it would leave a trace even after deletion.

**Comment by gildas-lormeau** [comment]
The auto-settings are made for this:
- Open the options page
- Click on "Auto-settings rules"
- Click on "display 'Auto-save profile' column" below the table
- In the table, add a rule like "regexp:github\.com", select "Disabled" as "Auto-save" profile (2nd column), and none of the pages on Github will be auto-saved

**Comment by mpeteuil** [comment]
Thanks for the quick reply @gildas-lormeau!

I tried to look for any mention of something like this before opening this issue but I couldn't find anything in the sources I checked, which were the documentation on the repository and the FAQ. I did see the "Auto-settings Rules" section in the extension preferences but it wasn't clear to me what that was and I didn't realize there was the help section (`extension/ui/pages/help.html`) of the extension until just now 😅

Your comment above + the help page should take care of things for me, thanks!
